### ISSUE TITLE
static interception delayMs option measures time from "cy.intercept" command, not from the request timestamp

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#14511]
Cypress v6.2.1 using `cy.intercept` to send a static response. My goal is to clearly see the loading indicator in the UI, thus I want to slow down the server response.

## Situation

Without intercept, the request fires 3 seconds after the page loads

```js
it('use positive then negative assertion', () => {
  cy.visit('/?delay=3000')
  // first, make sure the loading indicator shows up (positive assertion)
  cy.get('.loading').should('be.visible')
  // then assert it goes away (negative assertion)
  cy.get('.loading').should('not.be.visible')
})
```

Notice how quickly the loading message appears and disappears without network control. In fact it passes on the first run, and fails to "catch" the quickly appearing loading element on the second re-run.

https://user-images.githubusercontent.com/2212006/104367851-5bd91a80-54e9-11eb-985e-3512a4f7a54b.mp4

Of course, this is a flakey test and we could use test retries to solve it. But let's make the test better using network control.

## Attempt

So I want to slow down the network request to `GET /todos` to always have the loading indicator stay visible for a period of time.

```js
it('slows down the network response (does not work)', () => {
  cy.intercept('/todos', {
    body: [],
    delayMs: 2000
  })
  cy.visit('/?delay=3000')
  // first, make sure the loading indicator shows up (positive assertion)
  cy.get('.loading').should('be.visible')
  // then assert it goes away (negative assertion)
  cy.get('.loading').should('not.be.visible')
})
```

Hmm, it looks wrong! It looks like the `delayMs` wasn't even applied

https://user-images.githubusercontent.com/2212006/104368454-149f5980-54ea-11eb-860a-44181c1246b5.mp4

## Code

Looking at the code in https://github.com/cypress-io/cypress/blob/f43f1c1f26290fc8387697c8122f0f50b6c7cb55/packages/driver/src/cy/net-stubbing/static-response-utils.ts#L106

I see that we compute the response time using the time when `cy.intercept()` command runs

```js
if (staticResponse.delayMs) {
  backendStaticResponse.continueResponseAt = Date.now() + staticResponse.delayMs
}
```

Ok, let's see. If our application makes the request after 3 seconds, let's set the response to be sent after 5 seconds

https://user-images.githubusercontent.com/2212006/104368736-795ab400-54ea-11eb-923c-ecd0ea60d6fc.mp4

yeah, this works, but is so counterintuitive

## Workaround

Use programmatic reply and not a static response

```js
it('slows down the network response (programmatic)', () => {
  cy.intercept('/todos', req => {
    req.reply({
      body: [],
      delayMs: 2000
    })
  })
  cy.visit('/?delay=3000')
  // first, make sure the loading indicator shows up (positive assertion)
  cy.get('.loading').should('be.visible')
  // then assert it goes away (negative assertion)
  cy.get('.loading').should('not.be.visible')
})
```

BUT this solution also has a bug noted in #14446 - it uses the first time `req.reply` runs to "remember" the response delay :)

## Desired output

when using static intercept, start counting the delay from the moment the network request starts

**Comment by flotwig** [comment]
Duplicate of #14446

**Comment by jennifer-shehane** [comment]
Released in [`6.4.0`](https://on.cypress.io/changelog#6-4-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v6.4.0, please open a new issue.
